src/Riok.Mapperly/Descriptors/WellKnownTypes.cs

```diff
@@ -17,6 +17,13 @@ internal WellKnownTypes(Compilation compilation)
 
     public INamedTypeSymbol? TimeOnly => TryGet("System.TimeOnly");
 
+    public IArrayTypeSymbol GetArrayType(ITypeSymbol type) =>
+        _compilation.CreateArrayTypeSymbol(type, elementNullableAnnotation: type.NullableAnnotation);
+
+    public bool IsSpanType(ITypeSymbol symbol) =>
```

Comment:
To be discussed: this information could instead probably resolved via the `CollectionInfo`...

Reply:
This is only used in `EnsureCapacityBuilder.TryGetNonEnumeratedCount`. One of the main functions doesn't take `CollectionInfo` as a parameter.

Reply:
Its only used once in `BuildSpanToEnumerable`. I can use `TryBuildMapping` to generate it instead.